tools/dependency/pip_check.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python3
+
+import os
+import sys
+
+import yaml
+
+
+DEPENDABOT_CONFIG = ".github/dependabot.yml"
+
+
+def run_pip_check():
+
+    with open(os.path.join(sys.argv[1], DEPENDABOT_CONFIG)) as f:
+        dependabot = yaml.safe_load(f.read())
+
+    requirements_config = set(
+        update['directory']
+        for update in dependabot["updates"]
+        if update["package-ecosystem"] == "pip")
+
+    requirements_files = set()
+
+    for root, dirs, files in os.walk(sys.argv[1]):
+        requirements_files = (
+            requirements_files
+            | set(root.replace(sys.argv[1], "")
+                  for filename in files
+                  if filename == "requirements.txt"))
```

Comment:
as these are actually defined in bazel/repositories_extra.bzl it might be better to use a bazel query -  ie the name requirements.txt is kinda arbitrary

the problem with using a bazel query - is that not all requirements are/were defined in bazel but all files *are* called requirements.txt - so this works for now